tangelo/toolboxes/molecular_computation/molecule.py

```diff
@@ -404,34 +442,71 @@ def _convert_frozen_orbitals(self, frozen_orbitals):
 
         # First case: frozen_orbitals is an int.
         # The first n MOs are frozen.
-        if isinstance(frozen_orbitals, int):
+        if isinstance(frozen_orbitals, (int, np.integer)):
             frozen_orbitals = list(range(frozen_orbitals))
+            if self.uhf:
+                frozen_orbitals = [frozen_orbitals, frozen_orbitals]
         # Second case: frozen_orbitals is a list of int.
         # All MOs with indexes in this list are frozen (first MO is 0, second is 1, ...).
         # Everything else raise an exception.
-        elif not (isinstance(frozen_orbitals, list) and all(isinstance(_, int) for _ in frozen_orbitals)):
-            raise TypeError("frozen_orbitals argument must be an (or a list of) integer(s).")
-
-        occupied = [i for i in range(self.n_mos) if self.mo_occ[i] > 0.]
-        virtual = [i for i in range(self.n_mos) if self.mo_occ[i] == 0.]
+        elif isinstance(frozen_orbitals, list):
+            if self.uhf and not (len(frozen_orbitals) == 2 and all(isinstance(_, (int, np.integer)) for _ in frozen_orbitals[0]) and
+                                 all(isinstance(_, (int, np.integer)) for _ in frozen_orbitals[0])):
```

Comment:
This looks like an extra copy-paste. This line is difficult to parse but that `all(isinstance(_, (int, np.integer)) for _ in frozen_orbitals[0]` may not be necessary.